packages/cli/src/utils/generate.ts

```diff
@@ -0,0 +1,160 @@
+import fse from 'fs-extra'
+import fs from 'fs'
+import deepmerge from 'deepmerge'
+
+import {
+  CMSCoreDir,
+  CMSCustomDir,
+  CMSTmpDir,
+  coreDir,
+  customizationsDir,
+  customSrcDir,
+  repoDir,
+  storeConfigCoreFileDir,
+  storeConfigFileDir,
+  storeConfigTmpFileDir,
+  themeCustomizationsFileDir,
+  themesFileDir,
+  tmpDir,
+  tmpFolderName,
+} from './directory'
+
+interface GenerateOptions {
+  setup?: boolean
+}
+
+const ignoredDirs = ['node_modules']
+
+async function importStoreConfig() {
+  const { default: storeConfig } = await import(`${repoDir}/store.config.js`)
+  return storeConfig
+}
+
+function createTmpFolder() {
+  try {
+    fse.mkdirsSync(tmpDir)
+  } catch (err) {
+    console.error(err)
+  } finally {
+    console.log(`Temporary folder ${tmpFolderName} created`)
+  }
+}
+
+function copyCoreFiles() {
+  try {
+    fse.copySync(coreDir, tmpDir, {
+      filter(src) {
+        const splittedSrc = src.split('/')
+        return ignoredDirs.indexOf(splittedSrc[splittedSrc.length - 1]) === -1
```

Comment:
Just to make it easier to read (assuming `ignoredDirs` was renamed):

```suggestion
        const fileOrDirName = src.split('/').pop()
        const shouldCopy = !ignorePaths.includes(fileOrDirName)
        
        return shouldCopy
```